src/query/test/testdata/operators.test

```diff
@@ -188,15 +188,15 @@ eval instant at 50m (http_requests{group="canary"} + 1) and ignoring(group, job)
 #	vector_matching_a{l="x"} 10
 #	vector_matching_a{l="y"} 20
 
-# FAILING issue 34#. eval instant at 50m http_requests{group="canary"} unless http_requests{instance="0"}
-#	http_requests{group="canary", instance="1", job="api-server"} 400
-#	http_requests{group="canary", instance="1", job="app-server"} 800
+eval instant at 50m http_requests{group="canary"} unless http_requests{instance="0"}
+	http_requests{group="canary", instance="1", job="api-server"} 400
+	http_requests{group="canary", instance="1", job="app-server"} 800
 
 # FAILING issue #35. eval instant at 50m http_requests{group="canary"} unless on(job) http_requests{instance="0"}
 
-# FAILING issue #34. eval instant at 50m http_requests{group="canary"} unless on(job, instance) http_requests{instance="0"}
```

Comment:
What about other `#34` cases that we have? If they don't get fixed by this change, might mean that they are a different issue.

Reply:
+1, looks like we should be able to have the same fix in `or.go`, also looks like #23 might be solved by the same fix in `and.go`